### PR TITLE
Replace .unwrap() with ? in std::os::unix::net

### PR DESCRIPTION
**Pull request by linkmauve**
As people like to copy examples, this gives them good habits.

### CODE REVIEW

**Comment by Dylan-DPC** [comment]
hi @linkmauve thanks for the PR. The tests fail because we now need to return `Ok(())` in each of the functions

File: src/libstd/sys/unix/ext/mod.rs [hunk 1]
```diff
@@ -16,7 +16,7 @@
 //! use std::os::unix::prelude::*;
 //!
 //! fn main() {
```
Comment (hellow554) [inline src/libstd/sys/unix/ext/mod.rs]: Same with this example. Return type and return values are missing

File: src/libstd/sys/unix/ext/net.rs [hunk 2]
```diff
@@ -403,10 +427,13 @@ impl UnixStream {
...
     ///
-    /// let socket = UnixStream::connect("/tmp/sock").unwrap();
-    /// let result = socket.set_read_timeout(Some(Duration::new(0, 0)));
-    /// let err = result.unwrap_err();
-    /// assert_eq!(err.kind(), io::ErrorKind::InvalidInput)
+    /// fn main() -> std::io::Result<()> {
+    ///     let socket = UnixStream::connect("/tmp/sock")?;
+    ///     let result = socket.set_read_timeout(Some(Duration::new(0, 0)));
+    ///     let err = result.unwrap_err();
+    ///     assert_eq!(err.kind(), io::ErrorKind::InvalidInput)
```
Comment (hellow554) [inline src/libstd/sys/unix/ext/net.rs]: You are missing a semicolon here

File: src/libstd/sys/unix/ext/net.rs [hunk 3]
```diff
@@ -442,10 +473,13 @@ impl UnixStream {
...
     ///
-    /// let socket = UdpSocket::bind("127.0.0.1:34254").unwrap();
-    /// let result = socket.set_write_timeout(Some(Duration::new(0, 0)));
-    /// let err = result.unwrap_err();
-    /// assert_eq!(err.kind(), io::ErrorKind::InvalidInput)
+    /// fn main() -> std::io::Result<()> {
+    ///     let socket = UdpSocket::bind("127.0.0.1:34254")?;
+    ///     let result = socket.set_write_timeout(Some(Duration::new(0, 0)));
+    ///     let err = result.unwrap_err();
+    ///     assert_eq!(err.kind(), io::ErrorKind::InvalidInput)
```
Comment (hellow554) [inline src/libstd/sys/unix/ext/net.rs]: You are missing a semicolon here

File: src/libstd/sys/unix/ext/net.rs [hunk 4]
```diff
@@ -1319,10 +1417,13 @@ impl UnixDatagram {
...
     ///
-    /// let socket = UnixDatagram::unbound().unwrap();
-    /// let result = socket.set_read_timeout(Some(Duration::new(0, 0)));
-    /// let err = result.unwrap_err();
-    /// assert_eq!(err.kind(), io::ErrorKind::InvalidInput)
+    /// fn main() -> std::io::Result<()> {
+    ///     let socket = UnixDatagram::unbound()?;
+    ///     let result = socket.set_read_timeout(Some(Duration::new(0, 0)));
+    ///     let err = result.unwrap_err();
+    ///     assert_eq!(err.kind(), io::ErrorKind::InvalidInput)
```
Comment (hellow554) [inline src/libstd/sys/unix/ext/net.rs]: You are missing a semicolon here

File: src/libstd/sys/unix/ext/net.rs [hunk 5]
```diff
@@ -1359,10 +1463,13 @@ impl UnixDatagram {
...
     ///
-    /// let socket = UnixDatagram::unbound().unwrap();
-    /// let result = socket.set_write_timeout(Some(Duration::new(0, 0)));
-    /// let err = result.unwrap_err();
-    /// assert_eq!(err.kind(), io::ErrorKind::InvalidInput)
+    /// fn main() -> std::io::Result<()> {
+    ///     let socket = UnixDatagram::unbound()?;
+    ///     let result = socket.set_write_timeout(Some(Duration::new(0, 0)));
+    ///     let err = result.unwrap_err();
+    ///     assert_eq!(err.kind(), io::ErrorKind::InvalidInput)
```
Comment (hellow554) [inline src/libstd/sys/unix/ext/net.rs]: You are missing a semicolon here

File: src/libstd/sys/unix/ext/net.rs [hunk 6]
```diff
@@ -535,8 +582,11 @@ impl UnixStream {
     /// use std::os::unix::net::UnixStream;
     /// use std::net::Shutdown;
     ///
-    /// let socket = UnixStream::connect("/tmp/sock").unwrap();
-    /// socket.shutdown(Shutdown::Both).expect("shutdown function failed");
+    /// fn main() -> std::io::Result<()> {
+    ///     let socket = UnixStream::connect("/tmp/sock");
```
Comment (hellow554) [inline src/libstd/sys/unix/ext/net.rs]: You are missing a `?` here

**Comment by JohnCSimon** [comment]
Ping from triage:
@linkmauve can you address the comments from @hellow554 ?  Thanks!
cc: @cramertj

**Comment by JohnCSimon** [comment]
Pinging from triage:
@linkmauve can you address the comments from @hellow554 ? Also the build is failing. Thanks!
Thank you.

**Comment by linkmauve** [comment]
Thanks for the review, I just fixed the last few issues. :)

I also rebased it on master, maybe I shouldn’t have since it moved all of the commits to the end.

**Comment by JohnCSimon** [comment]
Ping from triage:
@cramertj - can you please review this PR?

**Pull request merged**